contents/blog/persons-on-events.md

```diff
@@ -0,0 +1,34 @@
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-array-blog.png
+featuredImageType: full
+---
+
+In [our last product update](/blog/the-posthog-array-1-39-0) we announced a beta for a substantial change to the way we handle persons and events on PostHog. Today, after gathering your feedback and seeing the improvements to query performance, we’re rolling this change out to all PostHog users. 
+
+This change essentially combines [persons](/manual/persons) and [events](/manual/events) into a single ClickHouse table, moving person IDs and properties _onto_ events. You won’t see any UI changes — persons will still have their own Persons & Groups section on the sidebar, for example — and there’ll be no loss of functionality, but here are some changes you may see:
+
+- **Faster results for queries involving person properties and events**. Putting persons and events into a single ClickHouse table means we no longer have to join tables to get results on queries involving these data. As a result, query performance will improve by up to 400%.
+
+- **Faster filtering of events with person properties**. Similarly, filtering events by person properties is much faster when a join is no longer necessary. Anywhere where you’re working with persons and events, PostHog will be faster. 
+
+- **Anonymous and identified users will no longer be merged**. Previously, when a user was identified through an event, such as logging in to a service, we would end up with separate records for unidentified and identified behavior. We would then [merge these into a single person](//docs/how-posthog-works/ingestion-pipeline#merging-two-persons). This could sometimes create conflicts and [other performance issues](/docs/how-posthog-works/ingestion-pipeline#consequences-of-merging). Avoiding this merge means some users may be double-counted, but is much more performant and mostly [mitigated by the event buffer](/docs/how-posthog-works/ingestion-pipeline/#1-event-buffer).
+
+- **You can create queries based on person properties at the time of an event.** This wasn’t previously possible and was often requested. Now, with this change, 
+- **Self-hosting users will see a storage increase**. This is due to extra information being stored in your self-hosted ClickHouse. We expect users to see their storage requirement increase by approximately XX. 
+
+Self-hosting users also need to manually upgrade deployment to take advantage of this change, while the change will be automatically targeted for users on PostHog Cloud. If you’re self-hosting PostHog, you need to run [async migration](/docs/runbook/async-migrations) number 0006. Afterwards, change your `PERSON_ON_EVENTS_ENABLED` [instance setting](/docs/self-host/configure/instance-settings) to `true`. You'll need to be [a Staff user](/docs/self-host/configure/instance-settings#staff-users) to make changes.
```

Comment:
The instance setting change depends on when this is going out. It will become automatic in 1.41 I think, so we want to message something like:

"You must run this migration before version 1.41". (cc: ingestion team for the accurate wording / plan for full release!)

Reply:
> to manually upgrade deployment

this makes it sound different from normal operational flow, I'd just say "need to update to the latest version of PostHog"

For the other part maybe sth like: This will be enabled with version 1.41, however if you'd like to check it out earlier you can do so by enabling the <flag info> after running the 0006 async migration.

Reply:
As we'll be holding this back for the full release, I think we can just cut the second part entirely. Thanks!